### PR TITLE
[UI] Streamline account settings

### PR DESCRIPTION
**Pull request by lachlanjc**
## Summary of the problem

- In the onboarding state, we had an overwhelming number of fields. It's pretty rare that users want to change their session duration or analytics settings, and they're not critical to getting started, so saved those for later.
- The notification options all had different copy and icons to refer to the same concepts, and had an unrelated setting for receipt importing in between. Grouping these as notifications and using consistent icons & copy makes them easier to parse.
- For our single-select tiles, having different widths & overflow onto two rows made the options look more overwhelming than they are.

(The receipt email generation is moving to Receipt Bin in #7662.)

## Describe your changes

| Before | After |
| ------ | ------ |
| Screen Shot 2024-08-22 at 18 52 17 | Screen Shot 2024-08-22 at 18 51 30 |
| image | image |

### CODE REVIEW

**Review by sampoder — DISMISSED** [review]
These look good to me! Let's get https://github.com/hackclub/hcb/pull/7662 merged and then this as well. Thanks for cleaning this up.

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -82,89 +83,91 @@
...
+        <fieldset class="field">
+          <%= form.label :receipt_report_option, "Receipt reminders" %>
+          <p class="h5 muted mt0 mb1">
+            Get a regular email roundup of your transactions awaiting receipts.
+          </p>
+          <%= form.select :receipt_report_option,
+                          (User.receipt_report_options.map { |k, v| [k.humanize, k] }) %>
+        </fieldset>
+        <fieldset class="field">
+          <legend class="font-medium">Transaction comments</legend>
```
Comment (garyhtou) [inline app/views/users/edit.html.erb]: just as an FYI, this setting also affects comments on other models such as Reimbursement Reports

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -82,89 +83,91 @@
...
             <% end %>
           </div>
         </fieldset>
         <fieldset>
-          <legend class="heading h2 pt2">Charge notifications</legend>
-          <p class="h5 muted mt1 mb2">
-            How would you like to receive charge notifications?
+          <legend class="font-medium">Card use</legend>
+          <p class="h5 muted mt-0 mb-3">
+            Get a notification when there’s a charge on your debit card.
```
Comment (garyhtou) [inline app/views/users/edit.html.erb]:
```suggestion
            Get a notification when there’s a charge on your HCB card.
```

**Review by garyhtou — COMMENTED** [review]
Thank you for this!

File: app/views/users/edit.html.erb (hunk 1 above)
Comment (sampoder) [reply app/views/users/edit.html.erb]: On yes, good catch, hence the threads wording

Comment (lachlanjc) [reply app/views/users/edit.html.erb]: Nice, added that information. "My Threads" I think wasn't clear enough, this should be clearer

Comment (sampoder) [reply app/views/users/edit.html.erb]: This is a minute detail but "my threads" also included threads you had previously commented on

**Comment by lachlanjc** [comment]
Glad you like! Can one of you hit approve?

**Review by sampoder — APPROVED** [review]
Let's merge in #7662 first so that people don't lose access to their mailbox address

**Pull request merged**